src/write/mod.rs

```diff
@@ -117,6 +119,26 @@ impl<'a> Object<'a> {
         self.architecture
     }
 
+    /// Return the sub-architecture.
+    #[inline]
+    pub fn sub_architecture(&self) -> Option<SubArchitecture> {
+        self.sub_architecture
+    }
+
+    /// Specify the sub-architecture.
+    pub fn set_sub_architecture(
+        &mut self,
+        sub_architecture: Option<SubArchitecture>,
+    ) -> Result<()> {
```

Comment:
Question for reviewers - does returning a `Result` here make sense? The alternative is to panic (`unimplemented!`), or to silently accept invalid sub-architectures.

Reply:
I'm inclined to silently accept here. We already need the check in `write` anyway, and that check takes into account whether the file format supports it. While this does delay the error, I think the message returned from `write` can be clear enough to make it easy to pinpoint.

Another option would be to use the type system to make it impossible to get wrong: instead of adding `SubArchitecture`, add `Aarch64SubArchitecture` and change `Architecture::Aarch64` to `Architecture::Aarch64(Aarch64SubArchitecture)`, and then there's no need for this `set_sub_architecture` at all. The disadvantage is this would be a breaking change every time we added support for sub-architectures to a new architecture, so I'm inclined not to do this.

Reply:
Ok, I'll move the check to `write`.

I also agree about `Architecture::Aarch64(Aarch64SubArchitecture)` - it's be nice, but I'm not sure it'd worth the SemVer bump (now and for each arch that gets a sub arch).